doc/Programming-guide/Your-first-Kaa-application/attach/demo-cpp/KaaDemo.cpp

```diff
@@ -25,22 +25,57 @@
 #include <vector>
 #include <list>
 
-// using namespace kaa;
-
-class Transceiver : public kaa::IConfigurationReceiver {
- private:
-    // Static variables
-    static constexpr auto savedConfig = "saved_config.cfg";
-    std::shared_ptr<kaa::IKaaClient> kaaClient;
-    int32_t samplePeriod;
-    boost::asio::io_service service;
-    boost::posix_time::seconds interval;
-    boost::asio::deadline_timer timer;
+class ConfigurationCollection : public kaa::IConfigurationReceiver {
+public:
+    ConfigurationCollection()
+        : kaaClient(kaa::Kaa::newClient())
+        , interval(samplePeriod = 0)
+        , timer(service, interval) 
+    {
+        // Set a custom strategy for uploading logs.
+        kaaClient->setLogUploadStrategy(
+            std::make_shared<kaa::RecordCountLogUploadStrategy>
+            (1, kaaClient->getKaaClientContext()));
```

Comment:
Add another indentation level, so it's clear `std::make_shared<kaa::RecordCountLogUploadStrategy>` and `(1, kaaClient->getKaaClientContext())` are related.